===================================================================
RCS file: /cvs/src/src/winsup/cygwin/include/cygwin/types.h,v
retrieving revision 1.33
retrieving revision 1.35
diff -u -r1.33 -r1.35
--- src/winsup/cygwin/include/cygwin/types.h	2011/03/29 10:32:40	1.33
+++ src/winsup/cygwin/include/cygwin/types.h	2011/04/05 00:53:17	1.35
@@ -9,15 +9,14 @@
 Cygwin license.  Please consult the file "CYGWIN_LICENSE" for
 details. */
 
+#ifndef _CYGWIN_TYPES_H
+#define _CYGWIN_TYPES_H
+
 #ifdef __cplusplus
 extern "C"
 {
 #endif
 
-#ifndef _CYGWIN_TYPES_H
-#define _CYGWIN_TYPES_H
-
-#include <sys/sysmacros.h>
 #include <stdint.h>
 #include <endian.h>
 
@@ -218,8 +217,12 @@
 /* semaphores types */
 typedef class semaphore *sem_t;
 #endif /* __INSIDE_CYGWIN__ */
-#endif /* _CYGWIN_TYPES_H */
+
+/* this header needs the dev_t typedef */
+#include <sys/sysmacros.h>
 
 #ifdef __cplusplus
 }
 #endif
+
+#endif /* _CYGWIN_TYPES_H */
===================================================================
RCS file: /cvs/src/src/winsup/cygwin/include/sys/sysmacros.h,v
retrieving revision 1.4
retrieving revision 1.5
diff -u -r1.4 -r1.5
--- src/winsup/cygwin/include/sys/sysmacros.h	2010/02/26 09:36:21	1.4
+++ src/winsup/cygwin/include/sys/sysmacros.h	2011/04/05 00:53:17	1.5
@@ -11,8 +11,30 @@
 #ifndef _SYS_SYSMACROS_H
 #define _SYS_SYSMACROS_H
 
-#define major(dev) ((int)(((dev) >> 16) & 0xffff))
-#define minor(dev) ((int)((dev) & 0xffff))
-#define makedev(major, minor) (((major) << 16) | ((minor) & 0xffff))
+_ELIDABLE_INLINE int gnu_dev_major(dev_t);
+_ELIDABLE_INLINE int gnu_dev_minor(dev_t);
+_ELIDABLE_INLINE dev_t gnu_dev_makedev(int, int);
+
+_ELIDABLE_INLINE int
+gnu_dev_major(dev_t dev)
+{
+	return (int)(((dev) >> 16) & 0xffff);
+}
+
+_ELIDABLE_INLINE int
+gnu_dev_minor(dev_t dev)
+{
+	return (int)((dev) & 0xffff);
+}
+
+_ELIDABLE_INLINE dev_t
+gnu_dev_makedev(int maj, int min)
+{
+	return (((maj) << 16) | ((min) & 0xffff));
+}
+
+#define major(dev) gnu_dev_major(dev)
+#define minor(dev) gnu_dev_minor(dev)
+#define makedev(maj, min) gnu_dev_makedev(maj, min)
 
 #endif /* _SYS_SYSMACROS_H */
